ral-facilities/datagateway
# Change Tests to Work with New Generator Script

## Description
This fixes #712. These changes make the e2e tests work with the updated icatdb generator script, changed as per https://github.com/ral-facilities/datagateway-api/issues/236. I've tried to use similar test data for each file as I noticed trends between files before these changes. For example, I've used `/browse/instrument/1/facilityCycle/16/investigation` quite a bit across different files.

I've also changed how the e2e tests for the admin download table work (https://github.com/ral-facilities/datagateway/commit/f121091181ced7564fa7808ff00c35e540089086 and https://github.com/ral-facilities/datagateway/commit/bef1f54f2ec7d41f0e1228eb1902607e1999a907). When looking for a prepared ID, it matches to a regex pattern instead of looking for a specific UUID as these will change for each ICAT instance. I've also tried to reduce the reliance on checking for prepared IDs (though not eliminated them) as I've noticed that even on fresh instances these can sometimes appear to be blank.

As preparation for the local CI work I've been doing, I've removed references to preprod URLs and now instead get the URLs for the settings files. There were some references not relating to e2e tests so I've also made these references look at the settings file. I've also done the same for a few facility names (mostly in `commands.js`) where I noticed relevant TODOs. I haven't removed all references to LILS just to avoid breaking things.

## Testing instructions
These have been tested while I've been working on the local CI, as can be seen passing [here](https://github.com/ral-facilities/datagateway/actions/runs/1037213161). To test these locally, you should repopulate your icatdb with `icatdb_generator.py` on master of DataGateway API. There are then a number of changes you need to make to your ICAT instance if not already done:

- Add `anon/anon` to rootUserNames in ICAT Server's `run.properties`
-  Disable two level storage on IDS by commenting out everything from the `# Properties for archive storage` section in `run.properties`, specifically: `plugin.archive.class`, `plugin.archive.dir`, `writeDelaySeconds`, `startArchivingLevel1024bytes`, `stopArchivingLevel1024bytes`, `storageUnit` and `tidyBlockSize`
- Reinstall ICAT Server and IDS Server after these changes
- Disable globus using `/topcat/admin/downloadType/globus/status`, using the message `"GLOBUS has been disabled for testing"`
- Recreate Lucene indexes for Datafile, Dataset and Investigation using `/icat/lucene/db/[ENTITY-NAME]/0`

Preprod's icatdb will need to be dropped and data regenerated using the fixed script.  There's a rough guide for it in one of the [API's CI jobs](https://github.com/ral-facilities/datagateway-api/blob/master/.github/workflows/ci-build.yml#L217). It would be great to get a dump (with the `--skip-comments` option) so I could check the generator script against that dump on the API's CI. I'm guessing this is something that @louise-davies and I will coordinate to do, but just writing here so I don't forget.

- [ ] Check Actions build (won't be relevant until preprod data has been changed or my local CI PR has been merged into this branch)
- [ ] Could someone look over the non-test files to make sure I haven't done anything silly? 

## Agile board tracking
connect to #712
